libfwupdplugin/fu-context.c

```diff
@@ -1442,21 +1476,14 @@ fu_context_get_esp_volumes(FuContext *self, GError **error)
 		g_autoptr(GPtrArray) devices = NULL;
 
 		/* udisks2 is working */
-		devices = fu_common_get_block_devices(NULL);
+		devices = fu_common_get_block_devices(error);
 		if (devices != NULL) {
 			g_set_error_literal(error,
 					    G_IO_ERROR,
 					    G_IO_ERROR_NOT_FOUND,
 					    "No ESP or BDP found");
 			return NULL;
 		}
```

Comment:
this all looks dodgy -- we're checking `devices != NULL`? I think we want devices ==  NULL to return without overwriting the error.

Reply:
Fixed